src/sqlancer/databend/DatabendSchema.java

```diff
@@ -0,0 +1,310 @@
+        // break;
+        // case "TIMESTAMP":
+        // primitiveType = DatabendDataType.TIMESTAMP;
+        // break;
+        case "NULL":
+            primitiveType = DatabendDataType.NULL;
+            break;
+        case "INTERVAL":
+            throw new IgnoreMeException();
+        // TODO: caused when a view contains a computation like ((TIMESTAMP '1970-01-05 11:26:57')-(TIMESTAMP
+        // '1969-12-29 06:50:27'))
+        default:
+            throw new AssertionError(typeString);
+        }
+        return new DatabendCompositeDataType(primitiveType, size);
+    }
+
+    public static class DatabendTable extends AbstractRelationalTable<DatabendColumn, TableIndex, DatabendGlobalState> {
+
+        public DatabendTable(String tableName, List<DatabendColumn> columns, boolean isView) {
+            super(tableName, columns, Collections.emptyList(), isView);
+        }
+
+    }
+
+    public static DatabendSchema fromConnection(SQLConnection con, String databaseName) throws SQLException {
+        List<DatabendTable> databaseTables = new ArrayList<>();
+        List<String> tableNames = getTableNames(con, databaseName);
+        for (String tableName : tableNames) {
+            if (DBMSCommon.matchesIndexName(tableName)) {
+                continue; // TODO: unexpected?
+            }
+            List<DatabendColumn> databaseColumns = getTableColumns(con, tableName, databaseName);
+            boolean isView = tableName.startsWith("v");
+            DatabendTable t = new DatabendTable(tableName, databaseColumns, isView);
+            for (DatabendColumn c : databaseColumns) {
+                c.setTable(t);
+            }
+            databaseTables.add(t);
+
+        }
+        return new DatabendSchema(databaseTables);
+    }
+
+    private static List<String> getTableNames(SQLConnection con, String databaseName) throws SQLException {
+        List<String> tableNames = null;
+        tableNames = new ArrayList<>();
+        // SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE table_schema != 'system' and table_schema !=
+        // 'INFORMATION_SCHEMA' and table_type='BASE TABLE'
+        // "SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE table_schema = '%s' and table_type='BASE TABLE' ",databaseName
+        final String sqlStatement = String.format(
+                "SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE table_schema = '%s' and table_type='BASE TABLE' ",
+                databaseName);
+        try (Statement s = con.createStatement()) {
+            try (ResultSet rs = s.executeQuery(sqlStatement)) {
+                try { // 没有catch的话rs.next()会报SQLException：Not a navigable ResultSet
+                    while (rs.next()) {
+                        tableNames.add(rs.getString("table_name"));
+                    }
+                } catch (Exception e) {
+                    // e.printStackTrace();
+                    System.out.println("TableNames->SQLException：Not a navigable ResultSet");
```

Comment:
Can this still happen? Is it expected? If not, perhaps we can just not catch the exception and quit execution (i.e., treating it as a bug).